### PR TITLE
Feature/display all cats

### LINKED ISSUE #2: As a user, I can navigate to a sidebar and select to see list of all cat breeds

### LINKED ISSUE #8: Use at least one external API

### LINKED ISSUE #9: Implement Typescript

### PR DESCRIPTION
**Pull request by shanekwarning**
### Description
* Summary:  This PR will create a new interface for our CatDetails component. It will allow us to start to use Router and NavLink to render single cats. 
* Where should the reviewer start: The reviewer will need to start in the app component. From there they can move to the cat details component. 

## Related Issues
Resolves #2 
Resolves #8 
Resolves #9 

### Type of change
Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -52,9 +54,14 @@ render() {
     <div className="App">
       <main>
         <h1> CATURDAY </h1>
-        {this.state.featuredCat && <CatDetails catInfo={this.state.featuredCat}/>}
-        <CatContainer />
-        <FeaturedCat catData={this.state.catData}/>
+         <Navbar />
```
Comment (angelewilliams) [inline src/App.tsx:57]: Nice job with routes

File: src/App.tsx [hunk 2]
```diff
@@ -52,9 +54,14 @@ render() {
...
+        <Route exact path='/cats' render={ () => <CatContainer catData={this.state.catData}/> } />
+       <Route exact path='/' render={() =>  <FeaturedCat catInfo={this.state.featuredCat}/> } />
```
Comment (emmacbd) [inline src/App.tsx:59]: indent here?

File: src/CatContainer.tsx [hunk 3]
```diff
@@ -1,12 +1,29 @@
...
+type CatData = {
+  catData: CatObject[] 
+}
+
+const CatContainer: React.FC<CatData> = ({catData}) => {
 
-const CatContainer = () => {
-  return (
-    <div>
-      <h2>hello cool cats and kittens </h2>
```
Comment (angelewilliams) [inline src/CatContainer.tsx:8]: Maybe Capitalize these? 🤪

File: src/CatDetails.tsx [hunk 4]
```diff
@@ -1,24 +1,60 @@
...
+// We now want to pass just the cat ID to fetch the correct cat. 
+// We want to store the cat object in state so that we can render the correct cat. 
+//we can not 
+const CatDetails: React.FC<CatProps>= ({catId}) => {
+  const[CatInfo , getCat ] = useState<SelectedCat[]>([])
+  
+  // const getKitten = async (catId:string) => {
+  //   const url = `https://api.thecatapi.com/v1/images/search?breed_ids=${catId}`;
+
+  //   try{
```
Comment (angelewilliams) [inline src/CatDetails.tsx:19]: Glad we tried this even though we did not end up using it.

File: src/CatDetails.tsx [hunk 5]
```diff
@@ -1,24 +1,60 @@
...
-      <h3>{catInfo.name}</h3>
-      <h3>{catInfo.temperament}</h3>
-      <h3>{catInfo.id}</h3>
-      {catInfo.image && <img className="cat-image" src={catInfo.image.url} alt={catInfo.name}/>}
-    </div> : ''
+// We now want to pass just the cat ID to fetch the correct cat. 
+// We want to store the cat object in state so that we can render the correct cat. 
+//we can not 
+const CatDetails: React.FC<CatProps>= ({catId}) => {
+  const[CatInfo , getCat ] = useState<SelectedCat[]>([])
```
Comment (angelewilliams) [inline src/CatDetails.tsx:14]: Cool to start learning Hooks here

**Review by angelewilliams — APPROVED** [review]

File: src/CatContainer.tsx [hunk 6]
```diff
@@ -1,12 +1,29 @@
...
-      <CatCards />
+  const catTiles = catData.map(cat => {
+    return (
+      <NavLink to={`/cats/${cat.id}`}>
+      <CatCards
+      id={cat.id}
```
Comment (emmacbd) [inline src/CatContainer.tsx:16]: Maybe indent these props for readability?

File: src/CatDetails.tsx [hunk 7]
```diff
@@ -1,24 +1,60 @@
...
+    fetch(`https://api.thecatapi.com/v1/images/search?breed_ids=${catId}`)
+    .then(response => response.json())
+    // .then(cat => {
+    //   return cat as SelectedCat[]
+    // })
+    .then(data => getCat(data))
+  }, [])
+
+  let displayCat = () => {
+    if(!CatInfo.length){
```
Comment (emmacbd) [inline src/CatDetails.tsx:37]: Nice job figuring this out Shane!

**Review by emmacbd — COMMENTED** [review]
Well done everyone! Thanks for all the driving Shane.

**Pull request merged**